mz_bokeh_package/utilities/auth.py

```diff
@@ -2,58 +2,37 @@
 This module takes care of the authentication of calls to bokeh apps
 
 """
-
-import requests
-import os
-from bokeh.io import curdoc
 from tornado.web import RequestHandler
-from typing import Optional
 
 from mz_bokeh_package.utilities.environment import Environment
+from mz_bokeh_package.utilities.graphql_api import GraphqlQueryError
+from mz_bokeh_package.utilities.current_user import CurrentUser
 
 
-def get_user(request_handler: RequestHandler) -> str:
+def get_user(request_handler: RequestHandler) -> bool | None:
     """
-    authenticate user based on api_key and user_key that are sent via the query parameters of the request and
+    authenticate user based on the api_key that is sent via the query parameters of the request and
     return the user_id if the user is authenticated and otherwise return None
 
     Args:
         request_handler: a tornado RequestHandler object that contains the query parameters of the request,
-            which contains the api_key and user_key of the user
+            which contains the api_key of the user
 
     Returns:
-        the user id when authentication is successful, and otherwise None
+        True when authentication is successful, and otherwise None
     """
 
     # bypass authentication in the case of health.py dashboard to allow GCP to perform health checks
     if request_handler.request.path.split("/")[-1] in ("health", "error"):
-        return "ok"
-
-    query_arguments = request_handler.request.query_arguments
-
-    # get the api_key from the request header
-    api_keys = query_arguments.get("api_key")
-    if api_keys is not None and len(api_keys) == 1:
-        api_key = api_keys[0]
-    else:
-        api_key = ""
-
-    # get the user_key from the request header
-    user_keys = query_arguments.get("user_key")
-    if user_keys is not None and len(user_keys) == 1:
-        user_key = user_keys[0]
-    else:
-        user_key = ""
+        return True
 
-    # in the development environment, allow overriding the api_key and user_key via env variables
-    if Environment.get_environment() == 'dev':
-        api_key = os.getenv('API_KEY', api_key)
-        user_key = os.getenv('USER_KEY', user_key)
+    api_key = request_handler.request.query_arguments.get("api_key")
```

Comment:
I tested this, and there are two issues here:
1. What you get back is actually a list with one element which is the API Key, so you need to add the same logic we currently have in the `CurrentUser.get_api_key` method for retrieving the API key from the query (which makes sense, since you are retrieving it from the query, just not via curdoc but via the request handler).
2. After you do that, you will see that what you get is a binary string and not a string, and when you send that to graphql, it will fail. If you look at the GraphQL module in the ingester and in the analysis apps, you will see the following lines of code:
   ```
   # FIXME: move this to the CurrentUser.get_api_key() method in the mz-bokeh-package
   if isinstance(api_key, bytes):
      api_key = api_key.decode('utf8')
   ```
   This code is what converts from a bit string to a string, and now is a good time to do this "FIXME". You are going to need to do this conversion in two places: in the `get_user` method in `auth.py` and in `CurrentUser.get_api_key`, since these are the two places where you are fetching the api key from the header. Since we want to avoid code duplication, then I suggest you implement a new static method in `CurrentUser`:
   ```
   def get_api_key_from_query_arguments(query_arguments: list) -> str | None:
      api_keys = query_arguments.get("api_key")
      if api_keys is not None and len(api_keys) == 1:
         api_key = api_keys[0]
         if isinstance(api_key, bytes):
            api_key = api_key.decode('utf8')
         return api_key
      else:
         return None 
   ```
   Then you can use this method in both `get_api_key` and `get_user`.